astroquery/ipac/irsa/most.py

```diff
@@ -0,0 +1,559 @@
+        obj_type = params.get("obj_type", False)
+        if not obj_type:
+            raise ValueError("When input type is 'mpc_input' key 'obj_type' is required.")
+        if obj_type not in ("Asteroid", "Comet"):
+            raise ValueError("Object type is case sensitive and must be one of: `Asteroid` or `Comet`")
+
+        if not params.get("mpc_data", False):
+            raise ValueError("When input type is 'mpc_input' key 'mpc_data' is required.")
+
+    def _validate_manual_input_type(self, params):
+        """
+        Validate required parameters when `input_type='manual_input'`.
+
+        Parameters
+        ----------
+        params : `dict`
+            Dictionary of query parameters to validate.
+
+        Raises
+        ------
+        ValueError
+            If the input does not have the minimum required parameters set to
+            an at least truthy value.
+        """
+        obj_type = params.get("obj_type", False)
+        if not obj_type:
+            raise ValueError("When input type is 'mpc_input' key 'obj_type' is required.")
+        if obj_type not in ("Asteroid", "Comet"):
+            raise ValueError("Object type is case sensitive and must be one of: `Asteroid` or `Comet`")
+
+        # MOST will always require at least the distance and eccentricity
+        # distance param is named differently in cases of asteroids and comets
+        if not params.get("eccentricity", False):
+            raise ValueError("When input_type is 'manual_input', 'eccentricity' is required.")
+
+        if obj_type == "Asteroid":
+            if not params.get("semimajor_axis", False):
+                raise ValueError("When obj_type is 'Asteroid', 'semimajor_axis' is required.")
+        elif obj_type == "Comet":
+            if not params.get("perih_dist", False):
+                raise ValueError("When obj_type is 'Comet', 'perih_dist' is required.")
+
+        # This seemingly can be whatever
+        if not params.get("body_designation", False):
+            params["body_designation"] = "Test"+params["obj_type"]
+
+    def _validate_input(self, params):
+        """
+        Validate the minimum required set of parameters, for a given input
+        type, are at least truthy.
+
+        These include the keys `catalog`, `input_type`, `output_mode` and
+        `ephem_step` in addition to keys required by the specified input type.
+
+        Parameters
+        ----------
+        params : `dict`
+            Dictionary of query parameters to validate.
+
+        Raises
+        ------
+        ValueError
+            If the input does not have the minimum required parameters set to
+            an at least truthy value.
+        """
+        if params.get("catalog", None) is None:
+            raise ValueError("Which catalog is being queried is always required.")
+
+        input_type = params.get("input_type", None)
+        if input_type is None:
+            raise ValueError("Input type is always required.")
```

Comment:
Sometimes, when they are the parameters that go into SPICE Toolkit, because SPICE has nice error messages. Other times you see unhelpful messages such as: https://github.com/astropy/astroquery/pull/2660#discussion_r1231377611